Migrate over to new terraform SDK

masutaka opened the pull request:
Per https://www.terraform.io/docs/extend/plugin-sdk.html.

It seemed to be a seamless upgrade.

* [x] TODO: When PR #15 is merged, I'll rebase this branch

kristofferahl commented:
@masutaka Let me know when the PR is rebased and I'll have a look. Great work my friend!

masutaka commented:
@kristofferahl Please review!

kristofferahl commented on go.mod:
```diff
@@ -3,6 +3,30 @@ module github.com/kristofferahl/terraform-provider-healthchecksio
+	google.golang.org/api v0.10.0 // indirect
+	google.golang.org/appengine v1.6.5 // indirect
+	google.golang.org/genproto v0.0.0-20191007204434-a023cd5227bd // indirect
+	google.golang.org/grpc v1.24.0 // indirect
```
Should indirect dependencies be included in go.mod? Asking because I don't know. :)

masutaka replied:
I may not understand `go get`. 4f079742a5ec7fa208390449ed03095fa205601d
I'm not sure, but it seems the `-u` option is not necessary. I will fix it later.

masutaka committed:
$ go get github.com/hashicorp/terraform

The reason is the below:

$ go mod vendor

$ tf-sdk-migrator check
Checking Go runtime version ...
Go version 1.13.1: OK.
Checking whether provider uses Go modules...
Go modules in use: OK.
Checking version of github.com/hashicorp/terraform-plugin-sdk to determine if provider was already migrated...
Checking version of github.com/hashicorp/terraform used in provider...
Terraform version does not satisfy constraint >=0.12.7. Found Terraform version: 0.12.2
Checking whether provider uses deprecated SDK packages or identifiers...
No imports of deprecated SDK packages or identifiers: OK.

Some constraints not satisfied. Please resolve these before migrating to the new SDK.

$ go get github.com/hashicorp/terraform

$ tf-sdk-migrator check
Checking Go runtime version ...
Go version 1.13.1: OK.
Checking whether provider uses Go modules...
Go modules in use: OK.
Checking version of github.com/hashicorp/terraform-plugin-sdk to determine if provider was already migrated...
Checking version of github.com/hashicorp/terraform used in provider...
Terraform version 0.12.10: OK.
Checking whether provider uses deprecated SDK packages or identifiers...
No imports of deprecated SDK packages or identifiers: OK.

All constraints satisfied. Provider can be migrated to the new SDK.

masutaka committed:
$ tf-sdk-migrator migrate

Checking Go runtime version ...
Go version 1.13.1: OK.
Checking whether provider uses Go modules...
Go modules in use: OK.
Checking version of github.com/hashicorp/terraform-plugin-sdk to determine if provider was already migrated...
Checking version of github.com/hashicorp/terraform used in provider...
Terraform version 0.12.10: OK.
Checking whether provider uses deprecated SDK packages or identifiers...
No imports of deprecated SDK packages or identifiers: OK.

All constraints satisfied. Provider can be migrated to the new SDK.

Rewriting provider go.mod file...
Rewriting SDK package imports...
Running `go mod tidy`...
Success! Provider is migrated to github.com/hashicorp/terraform-plugin-sdk v1.1.0.

It looks like this provider vendors dependencies. Don't forget to run `go mod vendor`.
Make sure to review all changes and run all tests.

masutaka commented:
Please review it again.

FYI: Many of the changes in go.sum are due to `go mod tidy`. It executed in `$ tf-sdk-migrator migrate`.

Commit:
- Syntax highlighting for terraform in readme (trick to fool Circle CI to build again).

Commit:
- Disabling docker image layer cache as it fails the build.

kristofferahl reviewed (approved):
Looks great @masutaka! Had to make some changes to the build in order to get it working (docker image layer caching is a paid feature it seems). Merging the PR!

kristofferahl commented:
@masutaka Should we create a new release?

masutaka commented:
Yes, it's good.

kristofferahl commented:
Created a new release with the tag v1.4.0.
https://github.com/kristofferahl/terraform-provider-healthchecksio/releases/tag/v1.4.0
Great work!
